src/main/java/io/vertx/serviceproxy/InterceptorHolder.java

```diff
@@ -0,0 +1,39 @@
+package io.vertx.serviceproxy;
+
+import io.vertx.core.Future;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.json.JsonObject;
+
+import java.util.function.Function;
+
+public class InterceptorHolder {
+
+  private String action;
```

Comment:
I think for type and configuration safety, this field should be final. An interceptor for a specific action probably is not expected to change to another kind of action during runtime. This will prevent malicious config changes too.

Reply:
Should it be set as **null** in case of InterceptorHolder constructor with no **action** parameter?

Reply:
yes, indeed, `null` should have the semantic meaning of `any`, no matter what kind of action always run it